iothub/device/src/Exceptions/IotHubClientException.cs

```diff
@@ -19,25 +20,43 @@ public class IotHubClientException : Exception
         [NonSerialized]
         private const string TrackingIdValueSerializationStoreName = "IotHubClientException-TrackingId";
 
-        private HashSet<IotHubStatusCode> transientStatusCodes = new HashSet<IotHubStatusCode>
+        private HashSet<IotHubClientErrorCode> transientErrorCodes = new HashSet<IotHubClientErrorCode>
         {
-            IotHubStatusCode.QuotaExceeded,
-            IotHubStatusCode.ServerError,
-            IotHubStatusCode.ServerBusy,
-            IotHubStatusCode.Throttled,
-            IotHubStatusCode.Timeout,
-            IotHubStatusCode.NetworkErrors,
+            IotHubClientErrorCode.QuotaExceeded,
+            IotHubClientErrorCode.ServerError,
+            IotHubClientErrorCode.ServerBusy,
+            IotHubClientErrorCode.Throttled,
+            IotHubClientErrorCode.Timeout,
+            IotHubClientErrorCode.NetworkErrors,
+        };
+
+        private Dictionary<IotHubClientErrorCode, HttpStatusCode> httpStatusCodes = new Dictionary<IotHubClientErrorCode, HttpStatusCode>()
+        {
+            { IotHubClientErrorCode.Ok, HttpStatusCode.OK },
+            { IotHubClientErrorCode.DeviceMaximumQueueDepthExceeded, HttpStatusCode.Forbidden },
+            { IotHubClientErrorCode.QuotaExceeded, HttpStatusCode.Forbidden },
+            { IotHubClientErrorCode.DeviceMessageLockLost, HttpStatusCode.PreconditionFailed },
+            { IotHubClientErrorCode.DeviceNotFound, HttpStatusCode.NotFound },
+            { IotHubClientErrorCode.NetworkErrors, HttpStatusCode.BadRequest },
```

Comment:
I feel like network error shouldn't be 400, but I'm not sure what code it should be.

Reply:
599? 408? 503?